src/H5Fpublic.h

```diff
@@ -20,56 +20,36 @@
 #include "H5ACpublic.h" /* Metadata Cache                           */
 #include "H5Ipublic.h"  /* Identifiers                              */
 
-/* When this header is included from a private header, don't make calls to H5check() */
-#undef H5CHECK
-#ifndef H5private_H
-#define H5CHECK H5check(),
-#else /* H5private_H */
-#define H5CHECK
-#endif /* H5private_H */
-
-/* When this header is included from a private HDF5 header, don't make calls to H5open() */
-#undef H5OPEN
-#ifndef H5private_H
-#define H5OPEN H5open(),
-#else /* H5private_H */
-#define H5OPEN
-#endif /* H5private_H */
-
 /*
  * These are the bits that can be passed to the `flags' argument of
  * H5Fcreate() and H5Fopen(). Use the bit-wise OR operator (|) to combine
- * them as needed.  As a side effect, they call H5check_version() to make sure
- * that the application is compiled with a version of the hdf5 header files
- * which are compatible with the library to which the application is linked.
- * We're assuming that these constants are used rather early in the hdf5
- * session.
- */
-#define H5F_ACC_RDONLY (H5CHECK H5OPEN 0x0000u) /**< Absence of RDWR: read-only */
```

Comment:
Having the H5OPEN macro in these definitions hasn't been necessary for a _long_ time.